ert_shared/main.py

```diff
@@ -405,13 +405,15 @@ def main():
     locale.setlocale(locale.LC_NUMERIC, "C")
 
     args = ert_parser(None, sys.argv[1:])
+    logger = logging.getLogger(__name__)
     if args.verbose:
-        logger = logging.getLogger()
         logger.setLevel("DEBUG")
     FeatureToggling.update_from_args(args)
-
-    with start_ert_server(), ErtPluginContext() as context:
-        context.plugin_manager.add_logging_handles(logging)
-        args.func(args)
-
+    try:
+        with start_ert_server(), ErtPluginContext() as context:
+            context.plugin_manager.add_logging_handles(logging)
```

Comment:
A chat with @sondreso has made me wonder why we are sending `logging` to `add_logging_handles` instead of a reference to the root-logger, i.e., the result of `logging.getLogger()`.

Do you recall the reasoning behind this?

Reply:
The short answer is it does not matter, and that is shown more explicitly this way. Logging is global, and because we hand over control to the plugin they are free to do `import logging` and do that for themselves if they want. We also make it easier to get a specific logger, and not just the root logger (which they can technically do anyway).

Reply:
The ert-azurelogger plugin currently expects the `logging`-module as input and adds a handler to `logging.getLogger()`, which I guess means that the handler is added to the root-logger.

Would it not be better to send a specific instance of the logger to the plugin so that it adds a handler to that specific instance instead?

Reply:
> The ert-azurelogger plugin currently expects the logging-module as input and adds a handler to logging.getLogger(), which I guess means that the handler is added to the root-logger.

Yes, but the plugin could add it with `logging.getLogger(__name__)` if wanted. And isnt that the flexibility we want, to be able to add it to the root logger if we want?

> Would it not be better to send a specific instance of the logger to the plugin so that it adds a handler to that specific instance instead?

Maybe, I dont quite see the argument as logging is usually done as:
```
import logging

logger = logging.getLogger()  # Or a name
```
and I feel this is the closest we get to that pattern.

Reply:
> > The ert-azurelogger plugin currently expects the logging-module as input and adds a handler to logging.getLogger(), which I guess means that the handler is added to the root-logger.
> 
> Yes, but the plugin could add it with `logging.getLogger(__name__)` if wanted. And isnt that the flexibility we want, to be able to add it to the root logger if we want?

Should we not let the host decide which logger the plugin should add handlers to?
If we call `getLogger(...)` from within the plugin, we let the plugin make the decision as to which logger the handler should be added.
This might however be a misunderstanding on my part...

Reply:
> Should we not let the host decide which logger the plugin should add handlers to?

We cant really do that, and think we would like the option of adding to the root logger anyway so we dont have to keep track of all the different loggers in `ert`. We might be talking past each other.

Reply:
Moved this discussion to: #2043